src/libostree/ostree-repo-pull.c

```diff
@@ -117,12 +117,6 @@ typedef struct {
 typedef struct {
   OtPullData  *pull_data;
   GVariant    *object;
-  gboolean     is_detached_meta;
-
-  /* Only relevant when is_detached_meta is TRUE.  Controls
-   * whether to fetch the primary object after fetching its
-   * detached metadata (no need if it's already stored). */
-  gboolean     object_is_stored;
```

Comment:
See the comment: "Only relevant when is_detached_meta is TRUE.". There were three combinations:
* is_detached_meta = FALSE: corresponds to a fetch of type COMMIT
* is_detached_meta = TRUE, object_is_stored = FALSE: corresponds to one fetch of type COMMIT and one of type COMMIT_META
* is_detached_meta = TRUE, object_is_stored = TRUE: corresponds to a fetch of type COMMIT_META

The code in scan_one_metadata_object_c prevents downloading commits multiple times, but commit_meta is always fetched. (There should probably be a test for that)